tests/unit/meta/test_meta.py

```diff
@@ -1204,6 +1204,46 @@ def test_generate_hook_wrappers(self):
                 ),
             )
 
+    @patch("snapcraft.internal.project_loader._config.Config.snap_env")
+    def test_generated_hook_wrappers_include_environment(self, mock_snap_env):
```

Comment:
Probably a good idea to add a test to ensure no environment gets inserted if classic.

Reply:
I added explicit coverage for _assemble_runtime_environment(), strict and classic.